src/game/objects/King/KingHit.js

```diff
@@ -15,10 +15,15 @@ class KingHit extends KingAnimated {
    */
   update(frame) {
     if (this.king.animation.isFinished) {
-      if (this.king.isOnGround) {
+      if (this.king.hp <= 0) {
+        return this.king.transitionState('dead').onUpdate(frame);
+      }
+      if (this.king.hp > 0 && this.king.isOnGround) {
         return this.king.transitionState('idle').onUpdate(frame);
       }
-      return this.king.transitionState('fall').onUpdate(frame);
+      if (this.king.hp > 0 && !this.king.isOnGround) {
+        return this.king.transitionState('fall').onUpdate(frame);
```

Comment:
Nie potrzebujesz w `if`ach tych `this.king.hp > 0`, bo gdyby było mniejsze/równe zero, to wszedłbyś w pierwszego `if`a, który `return`uje. (tzn. warunek nie jest potrzebny bo jest zawsze spełniony)